src/software/python_bindings/pass_generator.cpp

```diff
@@ -0,0 +1,92 @@
+#include "software/ai/passing/pass_generator.h"
+
+#include <pybind11/pybind11.h>
+
+#include "shared/parameter/cpp_dynamic_parameters.h"
+#include "software/ai/passing/eighteen_zone_pitch_division.h"
+#include "software/geom/rectangle.h"
+#include "software/python_bindings/pass_utilities.h"
+#include "software/python_bindings/python_binding_utilities.h"
+
+namespace py = pybind11;
+
+using EighteenZonePassGenerator  = PassGenerator<EighteenZoneId>;
+using EighteenZonePassEvaluation = PassEvaluation<EighteenZoneId>;
+
+/**
+ * Creates a PassGenerator<EighteenZoneId> with an EighteenZonePitchDivision created
+ * from the given world's field and a dict with the parameter overrides for
+ * PassingConfig.
+ *
+ * @param world the world
+ * @param passing_config_dict dict of overridden parameters for the PassingConfig.
+ * @return an EighteenZonePassGenerator
+ */
+EighteenZonePassGenerator createEighteenZonePassGenerator(const World& world,
+                                                          py::dict passing_config_dict)
+{
+    auto passing_config = std::make_shared<PassingConfig>();
+    updateDynamicParametersConfigFromDict(passing_config, passing_config_dict);
+
+    auto pitch_division = std::make_shared<EighteenZonePitchDivision>(world.field());
+
+    return EighteenZonePassGenerator(pitch_division, passing_config);
+}
+
+/**
+ * Gets the best pass for each of the 18 zones and returns them in a list of dicts
+ * as per pass_utilities.cpp.
+ * @param generator an EighteenZonePassGenerator
+ * @param world the world
+ * @return a list of dicts describing the best pass in each of the zones.
+ */
+py::list getBestPassesForAllZones(EighteenZonePassGenerator& generator,
```

Comment:
this can probably take any pass generator, see comment below

Reply:
function needs to be template of `ZoneEnum` type and so it can't accept any `PassGenerator`, but made the change anyways.